Sprint 1

grizzlygarillaz opened the pull request.

grizzlygarillaz committed:
make basic views

grizzlygarillaz committed:
deply to netlify

grizzlygarillaz committed:
deply to netlify

grizzlygarillaz committed:
add figma link | fix modules

grizzlygarillaz committed:
add figma link | fix modules

grizzlygarillaz committed:
fix routing error

grizzlygarillaz committed:
fix node version

grizzlygarillaz committed:
fix node version

grizzlygarillaz committed:
fix express to module type

grizzlygarillaz committed:
fix express, return to js type

grizzlygarillaz committed:
remove parcel

grizzlygarillaz committed:
remove express from package.json | set port 3000 at npm run start

grizzlygarillaz committed:
fix package.json | set port 3000 for netlify

grizzlygarillaz committed:
move devDependencies to dependecies

grizzlygarillaz committed:
trying fix netlify routing

grizzlygarillaz committed:
trying fix netlify routing

grizzlygarillaz committed:
fix 3000 port | add login page in chat list

dmitrymorozoff commented on .gitignore:
```diff
@@ -0,0 +1,6 @@
+/.parcel-cache
+/dist
+/node_modules
+package-lock.json
```
Для git важно наличие [пустой строки](https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log) в конце файла.

dmitrymorozoff replied:
Не стоит помещать package-lock.json файл в .gitignore, он необходим при установке модулей.

Подробнее можно почитать [тут](https://docs.npmjs.com/cli/v8/configuring-npm/packagpe-lock-json)

Хотя для тебя это не совсем актуально, так как используется yarn.lock. Но все равно лучше его отсюда убрать

dmitrymorozoff commented on package.json:
```diff
@@ -0,0 +1,24 @@
+  "description": "",
+  "scripts": {
+    "build": "parcel build src/index.html",
+    "start": "parcel build src/index.html && node server.js",
```
Здесь можно не дублировать скрипт `parcel build src/index.html`, а вызвать его таким образом 

`npm run build && node server.js`

dmitrymorozoff commented on package.json:
```diff
@@ -0,0 +1,24 @@
+  "devDependencies": {
+  },
+  "dependencies": {
+    "express": "^4.18.1",
```
Лучше задавать фиксированные версии зависимостям.

Если вы однажды решите, удалить `package-lock.json` файл и попытаться повторно установить зависимости с помощью команды `npm install`, npm установит последние версии зависимостей, что может привести к ошибкам. 

Плюс никогда не знаешь в какой версии появляется та или иная уязвимость, поэтому крайне рекомендую использовать статичную версию

dmitrymorozoff commented on server.js:
```diff
@@ -0,0 +1,10 @@
+const express = require('express')
+const PORT = 3000;
```
Часто требуется возможность получить порт снаружи скрипта, не изменяя код, это можно сделать через [переменное окружение](https://nodejs.org/dist/latest-v8.x/docs/api/process.html):

`process.env.PORT || PORT;`

dmitrymorozoff commented on src/components/chat.hbs:
```diff
@@ -0,0 +1,19 @@
+      <div class="chat-item__avatar"></div>
+      <span>Chat name</span>
+    </div>
+    <img class="chat__icon" src="{{icons.chat_settings}}" alt="">
```
Не нужно упускать из вида такую вещь как alt атрибуты у изображений. Эти атрибуты зачитывает скринридер. Так что их отсутствие сильно сказывается на доступности

dmitrymorozoff commented on src/components/chat_list.hbs:
```diff
@@ -0,0 +1,26 @@
+        </div>
+      </div>
+      <div class="chat-item__info">
+        <span class="chat-item__date">{{ this.date }}</span>
```
В html для даты и времени есть специальный тег time

dmitrymorozoff commented on src/style.sass:
```diff
@@ -0,0 +1,41 @@
+
+input, textarea
+  border: 0
+  outline: none
```
Лучше не отключать свойство outline. Его отключение сильно сказывается на [доступности](https://htmlacademy.ru/blog/boost/frontend/short-10), элемент перестаёт выделяться при фокусе.

dmitrymorozoff reviewed (commented):
Привет, хорошая качественная работа. Код аккуратный и чистый. По заданиям все в порядке. Пару комментариев по тем моментам которые можно улучшить. А так все в порядке.

grizzlygarillaz committed:
(.gitignore) remove package-lock
(.gitignore) add new line in the end
(package.json) replace a doubled parcel command with the same existing npm script
(package.json) change dependency versions to fixed
(.env) add global PORT variable
style(<img>) add alt attribute
style(<time>) replace <span> with datetime text to <time>
style(outlines) enable outlines

grizzlygarillaz committed:
(.gitignore) remove package-lock
(.gitignore) add new line in the end
(package.json) replace a doubled parcel command with the same existing npm script
(package.json) change dependency versions to fixed
(.env) add global PORT variable
style(<img>) add alt attribute
style(<time>) replace <span> with datetime text to <time>
style(outlines) enable outlines
